Lets blobs mine

cacogen opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

- Blobs have a 50% chance of destroying mineral turfs when hitting them, same as walls
- Adds balloon alerts to certain blob messages
- Adds contextual screen tips for blob overminds hovering over blob pieces
- Makes off-station blobs lighter than other blobs
- Fixes a couple of bugs

Check out this sick vid I made:

https://streamable.com/kly4a3

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game

Allows blobs to destroy mineral turfs when necessary, such as when spawning on Ice Box. 

Without this they face a distinct disadvantage against crew that can destroy mineral turfs with plasma cutters like a hot knife through butter to get a better angle on their core. 

The fact they can't already seems like an oversight left over from when they would never normally encounter them due to spawning on space stations.

The bubble alerts provide a lot of feedback that was previously missing and saves taking your eyes off the game to look at warning messages in the chat.

The screen tips save you having to read the help screed to at least know what two of the shortcuts are.

Making off-station blobs lighter makes it immediately obvious which blobs count towards victory.

<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog

<!-- If your PR modifies aspects of the game that can be concretely observed by players or admins you should add a changelog. If your change does NOT meet this description, remove this section. Be sure to properly mark your PRs to prevent unnecessary GBP loss. You can read up on GBP and it's effects on PRs in the tgstation guides for contributors. Please note that maintainers freely reserve the right to remove and add tags should they deem it appropriate. You can attempt to finagle the system all you want, but it's best to shoot for clear communication right off the bat. -->

:cl:
balance: Blobs can destroy mineral turfs like they can walls
qol: Adds balloon alerts for blob overmind
qol: Adds contextual screen tips for blob overmind
qol: Off-station blobs are lighter than regular blobs
fix: Fixes bug where on-station blobs were not counted towards victory when expanded from off-station blobs
fix: Fixes issue with cores and nodes having their color applied twice
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

cacogen committed:
Lets blobs mine

So they can destroy mineral turfs when necessary, such as after spawning on Ice Box. 

Without this they face a distinct disadvantage against crew that can destroy mineral turfs with plasma cutters like a knife through butter to get a better angle on their core.

They fact that they can't already seems like an oversight left over from when they would never normally encounter them due to spawning on stations in space.

Serxule commented:
i think this was disabled to prevent blobs from completely ignoring the station and just winning

Mothblocks commented on code/game/turfs/closed/minerals.dm:
```diff
@@ -176,6 +176,10 @@
 
+/turf/closed/mineral/blob_act(obj/structure/blob/B)
+	if(prob(50))
+		gets_drilled(null, FALSE)
```
Named param for `FALSE`, remove the `null` after.

cacogen commented:
>i think this was disabled to prevent blobs from completely ignoring the station and just winning

Well, blobs outside the station don't count towards winning. You also can't place factory or resource blobs  or relocate your core to a node there.

cacogen committed:
Adds balloon alerts for placing blobs off-station

cacogen committed:
Adds more balloon alerts

Serxule commented:
> > i think this was disabled to prevent blobs from completely ignoring the station and just winning
> 
> Well, blobs outside the station don't count towards winning. You also can't place factory or resource blobs or relocate your core to a node there.

this is probably just a leftover thing before that fix had come then

cacogen commented:
You can move your core into the middle of a mountain more easily now given you can just mine into it instead of having to find a cave but I'm not sure how valuable that is given it's very easy for you or your auto expanding blobs to accidentally trigger gibtonite which destroys a lot of your blob tiles and players can just use PkAs to mine around your blob and destroy the blobs surrounding your core with the extra damage PkAs do in low pressure.

cacogen committed:
Adds contextual screen tips for blob overmind blob interactions

cacogen committed:
This feels wrong

cacogen committed:
Spoils nice pristine code

cacogen committed:
Fixes bug where blobs were not counted towards victory when expanded from off-station

- Fixes issue with contextual screen tips appearing on blobs they shouldn't
- Makes off-station blobs visibly lighter

cacogen committed:
Remove blob screen tip shows up on all removable blobs

cacogen committed:
Fixes issue with cores and nodes having their color applied twice

cacogen committed:
Makes off-station blob lightening easier to distinguish

cacogen committed:
Fixes remove blob screentip once and for all

cacogen committed:
Lightens off-station blobs by half

D4C-420 commented:
Reminder that tram has red rock all over the place which the blob would be able to mine through. Could be interesting to see a blob might use this to their advantage given more often then not the rocks lead to space.

cacogen committed:
Fixes runtime and makes refund balloon alerts appear above the refunded blob

cacogen commented:
Seems like having to break through multiple layers of rock walls to reach space would be less worse for the crew than just being able to break a wall into space. 

It's also quite resource intensive to mine a lot of turfs, because it takes two resources per try and you only have a 50% chance to succeed.

cacogen committed:
Fixes mistake stopping it compiling and adds balloon alerts for upgrading blobs

cacogen committed:
Makes balloon alert for upgrading to reflective work and gives them desired grammar on the name

Mothblocks commented:
Surprised the ctxt screentips thing was that easy, super important QoL change

cacogen commented:
I'm trying to do it for AIs for a different PR in an attempt to legitimise the unatomised and presumptuous changes made to the screentips code here but it's not working out very well

![](https://i.imgur.com/MLtoPZA.png)

Mothblocks commented:
Put all Alt-LMB / Ctrl-LMB / Ctrl-Shift-LMB on the same line, I feel? Separate PR tho tbh

cacogen commented:
That is an easy workaround I like it

Mothblocks commented on code/game/atoms.dm:
```diff
@@ -2135,12 +2135,18 @@
 
+						// Alt-LMB
+						if (SCREENTIP_CONTEXT_ALT_LMB in context)
+							if (extra_context != "")
```
Can you do what I did for LMB/RMB? Ctrl-LMB and Alt-LMB should be on their own line, that's what the above comment meant.

It's all inlined because, y'know, super hot proc.

cacogen committed:
Puts Ctrl-LMB and Alt-LMB on one line
